client/src/containers/Gene/GeneActivitiesTable.js

```diff
@@ -73,26 +76,49 @@ class GeneActivitiesTable extends Component {
     )
   }
 
+  getActivityDescriptor = (activityItem, selfGeneId) => {
+    const entityKeys = [
+      'provenance/merged-from',
+      'provenance/merged-into',
+      'provenance/split-from',
+      'provenance/split-into',
+    ];
+    const selfEntityStub = {
+      'gene/id': selfGeneId,
+    };
+    const [entityKey] = entityKeys.filter((key) => (
+      activityItem[key] && activityItem[key]['gene/id'] === selfGeneId
+    ));
+    const [relatedEntityKey] = entityKeys.filter((key) => (
+      activityItem[key] && activityItem[key]['gene/id'] !== selfGeneId
+    ));
+
+    let eventType;
+    if (entityKey) {
+      if (activityItem['provenance/merged-into'] === activityItem[entityKey]) {
+        eventType = 'acquire_merge';
+      } else if (activityItem['provenance/merged-from'] === activityItem[entityKey]) {
+        eventType = 'merge_into';
+      } else if (activityItem['provenance/split-into'] === activityItem[entityKey]) {
+        eventType = 'split_from';
+      } else if (activityItem['provenance/split-from'] === activityItem[entityKey]) {
+        eventType = 'split_into';
+      }
+    } else {
+      eventType = activityItem['provenance/what'];
+    }
+
```

Comment:
As discussed on call, let's use "merged_from" and "merged_into" in the display.
Think it would be better here if the ActivityItem event type had a mapping defined somewhere, between code and label, rather than  having conditional code to produce `eventType`.
e.g:
```javascript
var some activityTypes = {"provenance/merged-from" : "Merged from", ...}`
....
eventType = activityTypes[entityKey];
```
Alternatively, the label could be stored in the DB, but not sure it belongs there, what do you think?

Reply:
The conditional is to identify the event role of the _other gene_. Say if the the gene profile that you are on is the "merge-into" gene of an event, the conditional will derive the role "merge-from" to describe the other gene. It's just some UI level logic, probably okay as is.